flow-nodes/api-builder-plugin-format-date/test/test.js

```diff
@@ -67,97 +80,68 @@ describe('flow-node format-date', () => {
 				const expectedMessage = 'Missing required parameter: date';
-				expect(result.args[1]).to.be.instanceOf(Error)
-					.and.to.have.property('message', expectedMessage);
-				expect(result.context).to.be.an('Object');
-				expect(result.context.error).instanceOf(Error)
+				expect(value).to.be.instanceOf(Error)
 					.and.to.have.property('message', expectedMessage);
 			});
 
 			it('should error when missing parameter - format', async () => {
-				// Invoke #formatDate with no format.
-				const flowNode = runtime.getFlowNode('format-date');
 
-				const result = await flowNode.formatDate({
+				const { value, output } = await flowNode.formatDate({
 					date: '2020-01-08T13:24:40+02:00',
 					format: null
 				});
-
-				expect(result.callCount).to.equal(1);
-				expect(result.output).to.equal('error');
-				expect(result.args[0]).to.equal(null);
+				expect(output).to.equal('error');
 				const expectedMessage = 'Missing required parameter: format';
-				expect(result.args[1]).to.be.instanceOf(Error)
+				expect(value).to.be.instanceOf(Error)
 					.and.to.have.property('message', expectedMessage);
-				expect(result.context).to.be.an('Object');
-				expect(result.context.error).instanceOf(Error)
-					.and.to.have.property('message', expectedMessage);
-
 			});
 		});
 
 		it('should format the input date when passed as string', async () => {
-			const flowNode = runtime.getFlowNode('format-date');
 
-			const result = await flowNode.formatDate({
+			const { value, output } = await flowNode.formatDate({
 				date: '2020-01-08T13:24:40+02:00',
 				format: 'L'
 			});
-
-			expect(result.callCount).to.equal(1);
-			expect(result.output).to.equal('next');
-			expect(result.args).to.deep.equal([null, '01/08/2020']);
-			expect(result.context).to.deep.equal({
-				date: '01/08/2020'
-			});
+			expect(output).to.equal('next');
+			expect(value).to.equal('01/08/2020');
 		});
 
 		it('should format the input date when passed as Date', async () => {
-			const flowNode = runtime.getFlowNode('format-date');
 
-			const result = await flowNode.formatDate({
+			const { value, output } = await flowNode.formatDate({
 				date: new Date('2020-01-08T13:24:40+02:00'),
 				format: 'L'
 			});
 
-			expect(result.callCount).to.equal(1);
-			expect(result.output).to.equal('next');
-			expect(result.args).to.deep.equal([null, '01/08/2020']);
-			expect(result.context).to.deep.equal({
-				date: '01/08/2020'
-			});
+			expect(output).to.equal('next');
+			expect(value).to.equal('01/08/2020');
 		});
 
 		/**
 		 * Check some of the available formatting that Moment.js provides.
 		 */
 		it('should format date as expected with different formats', async () => {
-			const flowNode = runtime.getFlowNode('format-date');
 
 			const testDate = '2020-01-08T13:24:40+02:00';
 			const testData = [
 				{
 					format: 'MMMM Do YYYY, h:mm:ss a',
-					formatted: 'January 8th 2020, 1:24:40 pm'
+					formatted: 'January 8th 2020, 1:24:40 am'
```

Comment:
this is unexpected

Reply:
very unexpected.  @deepakvk, please be careful of these kinds of changes, or at least, if you did the due-diligence check and discovered the reason for the difference, then it would be very helpful if you comment on your own PR to let us know why this is so.

now, as to _why_ it is different...  for starters, the tests fail for me.  tbh, I didn't run any of the tests when doing reviews, so I'll need to go back and do that again, but @deepakvk, don't forget to run tests.  What is your TZ on your laptop?  When I run the test, I get: "January 8th 2020, 11:24:40 am".

The date used is "2020-01-08T13:24:40+02:00".  That's 2h ahead of GMT (format is using UTC for some reason).  I would expect the time to be "11:24:40 am".

So, I think the test was wrong originally.  Now, how did @deepakvk end up with "1:24:40 am"?

Reply:
@wickedest , I am in Europe/Dublin timezone and I was getting 11:24:40 result same as you, but I have changed this because Mitko was getting the result 1:24:40 because he was on a different timezone. So I was confused which one to keep.

Reply:
Okay, so he's in +3, which makes his time 1pm.  It's a TZ issue.  Try [TimeZone-offset](https://momentjs.com/docs/#/manipulating/timezone-offset/)

Reply:
Resolved.